crypto/evp/pmeth_lib.c

```diff
@@ -563,7 +563,11 @@ int EVP_PKEY_CTX_get_signature_md(EVP_PKEY_CTX *ctx, const EVP_MD **md)
     if (!EVP_PKEY_CTX_get_params(ctx, sig_md_params))
         return 0;
 
-    tmp = EVP_get_digestbyname(name);
+    /*
+     * TODO(3.0): We need to pass the library context here - but at the time
+     * of writing we don't seem to store that in the EVP_PKEY_CTX.
+     */
+    tmp = evp_get_digestbyname_ex(NULL, name);
```

Comment:
Considering #10308, you could put in `ctx->libctx` and mark this `[Pending on #10308]`

Reply:
Rebased to pull in #10308, and fixed this.